src/t_mockiato.rs

```diff
@@ -0,0 +1,367 @@
+/// ```
+fn doctest() {}
+
+#[cfg(test)]
+mod t {
+
+use crate::{TestSuite, UniquelyOwned};
+use mockiato::*;
+
+struct Mockiato {}
+impl TestSuite for Mockiato {
+    const NAME: &'static str = "mockiato";
+
+    fn associated_types() {
+        // Traits are only allowed to contain methods
+        unimplemented!()
+    }
+
+    fn checkpoint() {
+        unimplemented!()
+    }
+
+    fn reference_parameters() {
+        #[mockable]
+        pub trait A {
+            fn foo(&self, x: &u32);
+        }
+
+        let mut mock = AMock::new();
+        mock.expect_foo(|x| x.partial_eq_owned(42)).returns(());
+
+        mock.foo(&42);
+    }
+
+    fn consume_parameters() {
+        // Mockiato can't pass any arguments, by clone or by move, to a return
+        // function, because it doesn't support return functions
+        unimplemented!()
+    }
+
+    fn consume_self() {
+        #[mockable]
+        pub trait A {
+            fn into_nothing(self);
+        }
+
+        let mut mock = AMock::new();
+        mock.expect_into_nothing();
+        mock.into_nothing();
+    }
+
+    fn external_trait() {
+        unimplemented!()
+    }
+
+    fn foreign() {
+        unimplemented!()
+    }
+
+    fn generic_method() {
+        // Only lifetimes are supported as generic parameters on methods
+        unimplemented!()
+    }
+
+    fn generic_return() {
+        #[mockable]
+        pub trait A<T> {
+            fn foo(&self) -> T;
+        }
+
+        let mut mock = AMock::<u32>::new();
+        mock.expect_foo().returns(42);
+        assert_eq!(42u32, mock.foo());
+    }
+
+    fn generic_struct() {
+        // Only traits can be made mockable
+        unimplemented!()
+    }
+
+    fn generic_trait() {
+        #[mockable]
+        pub trait A<T> {
+            fn foo(&self, t: T) -> u32;
+        }
+
+        let mut mock: AMock<i16> = AMock::new();
+        mock.expect_foo(|t| t.any()).returns(42);
+        assert_eq!(42, mock.foo(-1));
+    }
+
+    fn impl_trait() {
+        // Only traits can be made mockable , and traits may not
+        // `use -> impl Trait` syntax
+        unimplemented!()
+    }
+
+    fn inherited_trait() {
+        // According to the README, trait bounds are not supported
+        unimplemented!()
+     }
+
+    fn match_method() {
+        unimplemented!()
+    }
+
+    fn mock_struct() {
+        // Only traits can be made mockable
+        unimplemented!()
+    }
+
+    fn mock_trait() {
+        #[mockable]
+        pub trait A {
+            fn foo(&self, key: i16);
+        }
+
+        let _mock = AMock::new();
+    }
+
+    fn multi_trait() {
+        // Trait bounds are not supported, and custom derive is the only way to
+        // mock a trait, so there's no way for a mock to implement multiple
+        // traits.
+        unimplemented!()
+    }
+
+    fn return_call_with_args() {
+        unimplemented!()
+    }
+
+    fn return_reference() {
+        // Mockiato can only return 'static references
```

Comment:
This is only true when `#[mockable(static_references)]` is enabled.
Example of returning a non-static reference:
https://github.com/myelin-ai/mockiato/blob/master/tests/return_values_with_references_work.rs

Reply:
I see now.